public/app/plugins/panel/geomap/layers/data/geojsonLayer.ts

```diff
@@ -131,11 +125,24 @@ export const geojsonLayer: MapLayerRegistryItem<GeoJSONMapperConfig> = {
             return polyStyle(values);
           }
 
+          // Support styling polygons from Feature properties
+          const featureProps = feature.getProperties();
+          const styleStrings: string[] = Object.values(GeoJSONStyles);
```

Comment:
move this outside the inner loop -- currently it will be created for each feature

Reply:
Good catch!